### ISSUE TITLE
Connection refused when using dfs mutli-cpu in loops

### ISSUE THREAD

**Issue by ruifeng96150** [FeatureLabs/featuretools#281]
### Bug/Feature Request Description

sometimes, the datas and labels is very big, 
so it is very slow to generate the feature matrix.
I am trying to split those data to smaller ones.
It's run well at the first batch.
But at the second or followings. always get tornado errors
Hope your help and support , Thanks!!!

run script like this:

```
def getEs(i):
    data_patch,label_patch = getDataLabel(i)
    es = ft.EntitySet(id="Online Retail Logs")

    # Add the entire data table as an entity
    es.entity_from_dataframe("purchases",
                             dataframe=data_patch,
                             index="purchases_index",
                             time_index = 'time')
    es.normalize_entity(new_entity_id="user_items",base_entity_id="purchases",index="uis")
    feature_matrix,feature_names = ft.dfs(entityset=es, target_entity='user_items',
                                           cutoff_time = label_patch, verbose = 2,
                                           cutoff_time_in_index = True,
                                           chunk_size = 100, n_jobs = -1,
                                           max_depth = 1)

    feature_matrix = feature_matrix.reset_index()
    feature_matrix.to_csv('data/fm/%d.csv' % (i),index=False)
    return feature_matrix

for i in range(1,10):
   fm = getEs(i)
```

output like this:
```
Elapsed: 31:01 | Remaining: 00:00 | Progress: 100%|██████████| Calculated: 4526/4526 chunks  
tornado.application - ERROR - Future exception was never retrieved: Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tornado/gen.py", line 1141, in run
    yielded = self.gen.throw(*exc_info)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/distributed/comm/tcp.py", line 333, in connect
    convert_stream_closed_error(self, e)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/distributed/comm/tcp.py", line 124, in convert_stream_closed_error
    raise CommClosedError("in %s: %s: %s" % (obj, exc.__class__.__name__, exc))
CommClosedError: in <distributed.comm.tcp.TCPConnector object at 0x10ebb9950>: error: [Errno 61] Connection refused
tornado.application - ERROR - Future exception was never retrieved: Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tornado/gen.py", line 1141, in run
    yielded = self.gen.throw(*exc_info)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/distributed/comm/tcp.py", line 333, in connect
    convert_stream_closed_error(self, e)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/distributed/comm/tcp.py", line 124, in convert_stream_closed_error
    raise CommClosedError("in %s: %s: %s" % (obj, exc.__class__.__name__, exc))
CommClosedError: in <distributed.comm.tcp.TCPConnector object at 0x10eb1ab90>: error: [Errno 61] Connection refused

```

-----
*Issues created here on Github are for bugs or feature requests. For usage questions and questions about errors, please ask on Stack Overflow with the [featuretools](https://stackoverflow.com/questions/tagged/featuretools) tag. Check the [documentation](https://docs.featuretools.com/#get-help) for further guidance on where to ask your question.*

**Comment by kmax12** [comment]
we've come across this error before and suspect it relates to a race condition inside of Dask (which is the library we use for parallelism).

does your code still run to completion or does this error cause it to stop?

**Comment by ruifeng96150** [comment]
@kmax12 yes most of time, it can run normally without interrupt.
But sometimes it would fail to start a new loop with following error message:

```
raise gen.TimeoutError("Worker failed to start")
tornado.gen.TimeoutError: Worker failed to start
```
And sometimes it will hint some warning:

```
Elapsed: 00:00 | Remaining: ? | Progress:   0%|          | Calculated: 0/4475 chunks
distributed.core - INFO - Event loop was unresponsive in Nanny for 4.03s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Nanny for 4.03s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Scheduler for 4.03s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Nanny for 4.04s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Nanny for 4.04s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
Elapsed: 00:27 | Remaining: 1:06:42 | Progress:   1%|▏         | Calculated: 58/4475 chunks
Exception tornado.util.TimeoutError: TimeoutError('timed out after 20 s.',) in <bound method LocalCluster.__del__ of LocalCluster('tcp://127.0.0.1:64968', workers=0, ncores=0)> ignored
Exception tornado.util.TimeoutError: TimeoutError('timed out after 20 s.',) in <bound method LocalCluster.__del__ of LocalCluster('tcp://127.0.0.1:64968', workers=0, ncores=0)> ignored
Exception tornado.util.TimeoutError: TimeoutError('timed out after 20 s.',) in <bound method LocalCluster.__del__ of LocalCluster('tcp://127.0.0.1:64968', workers=0, ncores=0)> ignored
Elapsed: 00:47 | Remaining: 4:36:19 | Progress:   1%|▏         | Calculated: 60/4475 chunks
Exception tornado.util.TimeoutError: TimeoutError('timed out after 20 s.',) in <bound method LocalCluster.__del__ of LocalCluster('tcp://127.0.0.1:64968', workers=0, ncores=0)> ignored
Elapsed: 17:06 | Remaining: 3:20:23 | Progress:  41%|████▏     | Calculated: 1849/4475 chunks 
distributed.core - INFO - Event loop was unresponsive in Nanny for 16.21s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Nanny for 16.26s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Nanny for 16.28s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Scheduler for 16.30s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Nanny for 16.32s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
Elapsed: 29:00 | Remaining: 1:26:56 | Progress:  81%|████████  | Calculated: 3624/4475 chunks
distributed.core - INFO - Event loop was unresponsive in Nanny for 3.17s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Nanny for 3.20s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Nanny for 3.21s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Scheduler for 3.21s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Nanny for 3.21s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
Elapsed: 37:19 | Remaining: 00:00 | Progress: 100%|██████████| Calculated: 4475/4475 chunks  
distributed.scheduler - INFO - Clear task state
distributed.nanny - INFO - Closing Nanny at 'tcp://127.0.0.1:55322'

distributed.nanny - INFO - Closing Nanny at 'tcp://127.0.0.1:55324'
distributed.nanny - INFO - Closing Nanny at 'tcp://127.0.0.1:55323'
distributed.nanny - INFO - Closing Nanny at 'tcp://127.0.0.1:55325'
distributed.core - INFO - Event loop was unresponsive in Nanny for 3.34s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Nanny for 3.34s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Nanny for 3.25s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Scheduler for 3.25s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Nanny for 3.26s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.scheduler - INFO - Scheduler closing...
distributed.scheduler - INFO - Scheduler closing all comms
distributed.scheduler - INFO - Remove worker tcp://127.0.0.1:55326
distributed.core - INFO - Removing comms to tcp://127.0.0.1:55326
distributed.scheduler - INFO - Remove worker tcp://127.0.0.1:55328
distributed.core - INFO - Removing comms to tcp://127.0.0.1:55328
distributed.scheduler - INFO - Remove worker tcp://127.0.0.1:55332
distributed.core - INFO - Removing comms to tcp://127.0.0.1:55332
distributed.scheduler - INFO - Remove worker tcp://127.0.0.1:55330
distributed.core - INFO - Removing comms to tcp://127.0.0.1:55330
distributed.scheduler - INFO - Lost all workers
distributed.core - INFO - Event loop was unresponsive in Nanny for 6.14s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Nanny for 6.15s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Nanny for 6.30s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
distributed.core - INFO - Event loop was unresponsive in Nanny for 6.30s.  This is often caused by long-running GIL-holding functions or moving large chunks of data. This can cause timeouts and instability.
('i', 40000)
```

**Comment by kmax12** [comment]
@shayan113 can you run `featuretools info` and let us know your setup that is running into this issue?

also, an easy work around if you don't care about run time would be to set `n_job=1` when you calculate features

**Comment by rwedge** [comment]
May have been fixed by #853, if problem persists we can re-open
